File: src/main/scala/wordcount/WordCount.scala
package wordcount

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object WordCount extends App {
    val conf = new SparkConf().setMaster("local").setAppName("WordCount")
    val sc = new SparkContext(conf) 

    val in = "/Users/stuart/git/spark-get-started/src/main/resources/shakespeare.txt"
    val out = "/Users/stuart/git/spark-get-started/src/main/resources/shakespeare.counts"

    val shakespeare = sc.textFile(in)
    val shakespeareCounts = shakespeare.flatMap(line => line.split(" "))
        .map(word => word.replaceAll("[^a-zA-Z]", ""))
        .map(word => (word, 1))
        .reduceByKey{case (x, y) => x + y}

    shakespeareCounts.saveAsTextFile(out) 

    sc.stop()
    sys.exit()
}

File: src/main/python/wordcount/WordCount.py
# Create new SparkContext
from pyspark import SparkConf, SparkContext
conf = SparkConf().setMaster("local").setAppName("WordCount")
sc = SparkContext(conf = conf)

inFile = "../../resources/wordcount/shakespeare.txt"
outFile = "./counts"

# WordCount on shakespeare using map and reduce
shakespeare = sc.textFile(inFile)
shakespeareCounts = shakespeare.flatMap(lambda line: line.split(" ")) \
                                .map(lambda word: word.replace("[^a-zA-Z]", "")) \
                                .map(lambda word: (word, 1)) \
                                .reduceByKey(lambda a, b: a + b)

shakespeareCounts.saveAsTextFile(outFile)
